src/main/java/com/aol/cyclops2/internal/stream/spliterators/DistinctKeySpliterator.java

```diff
@@ -34,7 +34,7 @@ public void forEachRemaining(Consumer<? super T> action) {
         final Consumer<? super IN> toUse = apply(action);
         source.forEachRemaining(e->{
 
-            if(!values.add(keyExtractor.apply(e))){
+            if(values.add(keyExtractor.apply(e))){
```

Comment:
How does this change still comply with getting distinct values for lists with duplicates?